presto-main/src/main/java/com/facebook/presto/operator/aggregation/noisyaggregation/NoisyCountGaussianColumnAggregation.java

```diff
@@ -0,0 +1,168 @@
+    {
+        DynamicClassLoader classLoader = new DynamicClassLoader(NoisyCountGaussianColumnAggregation.class.getClassLoader());
+
+        AccumulatorStateSerializer<CountScaleState> stateSerializer = StateCompiler.generateStateSerializer(CountScaleState.class, classLoader);
+        AccumulatorStateFactory<CountScaleState> stateFactory = StateCompiler.generateStateFactory(CountScaleState.class, classLoader);
+        Type intermediateType = stateSerializer.getSerializedType();
+
+        List<Type> inputTypes = ImmutableList.of(type, DOUBLE);
+
+        AggregationMetadata metadata = new AggregationMetadata(
+                generateAggregationName(NAME, BIGINT.getTypeSignature(), inputTypes.stream().map(Type::getTypeSignature).collect(toImmutableList())),
+                createInputParameterMetadata(type),
+                INPUT_FUNCTION,
+                COMBINE_FUNCTION,
+                OUTPUT_FUNCTION,
+                ImmutableList.of(new AccumulatorStateDescriptor(
+                        CountScaleState.class,
+                        stateSerializer,
+                        stateFactory)),
+                BIGINT);
+
+        Class<? extends Accumulator> accumulatorClass = AccumulatorCompiler.generateAccumulatorClass(
+                Accumulator.class,
+                metadata,
+                classLoader);
+        Class<? extends GroupedAccumulator> groupedAccumulatorClass = AccumulatorCompiler.generateAccumulatorClass(
+                GroupedAccumulator.class,
+                metadata,
+                classLoader);
+        return new BuiltInAggregationFunctionImplementation(NAME, inputTypes, ImmutableList.of(intermediateType), BIGINT,
+                true, false, metadata, accumulatorClass, groupedAccumulatorClass);
+    }
+
+    private static List<ParameterMetadata> createInputParameterMetadata(Type type)
+    {
+        return ImmutableList.of(
+                new ParameterMetadata(STATE),
+                new ParameterMetadata(BLOCK_INPUT_CHANNEL, type),
+                new ParameterMetadata(BLOCK_INPUT_CHANNEL, DOUBLE),
+                new ParameterMetadata(BLOCK_INDEX));
+    }
+
+    public static void input(CountScaleState state, Block valueBlock, Block noiseScaleBlock, int index)
+    {
+        double noiseScale = DOUBLE.getDouble(noiseScaleBlock, index);
+        if (noiseScale < 0) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "Noise scale must be >= 0");
+        }
+        // Update count and retain scale and random seed
+        state.setCount(state.getCount() + 1);
+        state.setNoiseScale(noiseScale);
+    }
+
+    public static void combine(CountScaleState state, CountScaleState otherState)
+    {
+        state.setCount(state.getCount() + otherState.getCount());
+        state.setNoiseScale(state.getNoiseScale() > 0 ? state.getNoiseScale() : otherState.getNoiseScale()); // noise scale should be > 0
```

Comment:
won't the noise scales always be the same as each other (including that they could both be 0), since ultimately they come from the function input?

Reply:
In unittest assertAggregation create an *empty* state (with noiseScale = 0) and them combine that empty state with otherState coming from actual execution on data. So there is this check